### PR TITLE
Formula- and term-based Wald-tests

### PR DESCRIPTION
**Pull request by MartinStancsicsQC**
Checklist
* [x] Added a `CHANGELOG.rst` entry

This PR uses the new functionality in glum v3 (namely the formula interface and the existence of term names) to extend the interface for the `wald_test` method. The latter now also accepts a `term` or a `formula` parameter.

### CODE REVIEW

File: src/glum/_glm.py [hunk 1]
```diff
@@ -1796,6 +1842,191 @@ def _wald_test_feature_names(
...
+
+        Perform a Wald test for the hypothesis that the coefficients of the
+        features in ``terms`` are equal to the values in ``terms``.
+
+        Parameters
+        ----------
+        terms : Union[str, list[str]]
+            The name of a term or a list of terms to test.  In the case of a model based
+            on a formula, a term is one of the expressions separated by ``+`` signs.
+            Otherwise, a term is one column in the input data. In the case of
```
Comment (MatthiasSchmidtblaicherQC) [inline src/glum/_glm.py:1945]: See above.

Comment (MatthiasSchmidtblaicherQC) [reply src/glum/_glm.py:1945]: It is probably worth expanding in docstring of `wald_test` how categoricals that are never encoded as dummies are handled. This seems like a distinguishing feature of glum's wald tests.

**Review by MatthiasSchmidtblaicherQC — CHANGES_REQUESTED** [review]

File: src/glum/_glm.py [hunk 2]
```diff
@@ -1522,6 +1525,11 @@ def wald_test(
             to test.
         features : Union[str, list[str]], optional, default=None
             The name of a feature or a list of features to test.
+        terms : Union[str, list[str]], optional, default=None
+            The name of a term or a list of terms to test.  In the case of a model based
+            on a formula, a term is one of the expressions separated by ``+`` signs.
+            Otherwise, a term is one column in the input data. In the case of
+            categorical variables, it translates to possibly multiple features.
```
[the remark this answers is not in the document]
Comment (MartinStancsicsQC) [reply src/glum/_glm.py:1532]: Yes, exactly. I'll try to make it clearer.

[the remark this answers is not in the document]
Comment (MartinStancsicsQC) [reply src/glum/_glm.py]: How about [this](https://github.com/Quantco/glum/blob/89a5ba4ae7f9b7861288f59670b41c551c3f9cb5/src/glum/_glm.py#L1532-L1538)?

**Review by MarcAntoineSchmidtQC — APPROVED** [review]
Looks good and well tested. Happy to merge this when you are ready.

File: src/glum/_glm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MatthiasSchmidtblaicherQC) [reply src/glum/_glm.py]: Looks good!

**Review by MatthiasSchmidtblaicherQC — APPROVED** [review]

**Comment by MartinStancsicsQC** [comment]
Great, I'll merge it right after the formula PR is merged.

**Pull request merged**